spec/requests/api/cloud_volumes_spec.rb

```diff
@@ -40,4 +40,87 @@
       "id"   => cloud_volume.id
     )
   end
+
+  it "rejects delete request without appropriate role" do
+    api_basic_authorize
+
+    run_post(cloud_volumes_url, :action => 'delete')
+
+    expect(response).to have_http_status(:forbidden)
+  end
+
+  it "can delete a single cloud volume" do
+    zone = FactoryGirl.create(:zone, :name => "api_zone")
+    aws = FactoryGirl.create(:ems_amazon, :zone => zone)
+
+    cloud_volume1 = FactoryGirl.create(:cloud_volume, :ext_management_system => aws, :name => "CloudVolume1")
+
+    api_basic_authorize action_identifier(:cloud_volumes, :delete, :resource_actions, :post)
+
+    run_post(cloud_volumes_url(cloud_volume1.id), :action => "delete")
+
+    expected = {
+      'message' => 'Deleting Cloud Volume CloudVolume1',
+      'success' => true,
+      'task_id' => a_kind_of(Numeric)
+    }
+
+    expect(response.parsed_body).to include(expected)
+    expect(response).to have_http_status(:ok)
+  end
+
+  it "can delete a cloud volume with DELETE as a resource action" do
+    zone = FactoryGirl.create(:zone, :name => "api_zone")
+    aws = FactoryGirl.create(:ems_amazon, :zone => zone)
+
+    cloud_volume1 = FactoryGirl.create(:cloud_volume, :ext_management_system => aws, :name => "CloudVolume1")
+
+    api_basic_authorize action_identifier(:cloud_volumes, :delete)
+
+    run_delete cloud_volumes_url(cloud_volume1.id)
+
+    expect(response).to have_http_status(:no_content)
+  end
+
+  it "can delete a cloud volume with DELETE as a collection action" do
+    zone = FactoryGirl.create(:zone, :name => "api_zone")
+    aws = FactoryGirl.create(:ems_amazon, :zone => zone)
+
+    cloud_volume1 = FactoryGirl.create(:cloud_volume, :ext_management_system => aws, :name => "CloudVolume1")
+
+    api_basic_authorize action_identifier(:cloud_volumes, :delete)
```

Comment:
Similarly `action_identifier(:cloud_volumes, :delete, :collection_actions, :delete)` + test for rejection